x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_factory.ts

```diff
@@ -260,7 +260,9 @@ export const createSecurityRuleTypeFactory: CreateSecurityRuleTypeFactory = ({
         }
 
         if (result.warningMessages.length) {
-          const warningMessage = buildRuleMessage(result.warningMessages.join());
+          const warningMessage = buildRuleMessage(
+            truncateMessageList(result.warningMessages).join()
+          );
```

Comment:
`buildRuleMessage` appends some debug information to the `warningMessage`:
```ts
  [
    ...messages,
    `name: "${name}"`,
    `id: "${id}"`,
    `rule id: "${ruleId ?? '(unknown rule id)'}"`,
    `signals index alias: "${index}"`,
  ].join(' ');
  ```
  
 This info will later get truncated by `truncateMessage`, which doesn't look right. 
 
I think we should remove truncation logic from exec log client level and embed it into `buildRuleMessageFactory` instead.